### ISSUE TITLE
allowedComparisons ignores "and", "or" blacklist

### ISSUE THREAD

**Issue by renehauck** [doug-martin/nestjs-query#893]
**Describe the bug**
I have a DTO "News", this has a property `storeId` . Now I want to allow only one `eq query` in this DTO, but `and` and `or` are automatically added.

### Have you read the [Contributing Guidelines](https://github.com/doug-martin/nestjs-query/blob/master/CONTRIBUTING.md)?

yes

**To Reproduce**
1. Create DTO `News`
```typescript
@ObjectType('PublicNews')
export class PublicNewsDto extends PickType(NewsDto, ['title', 'message', 'id']) {
  @FilterableField({
    allowedComparisons: ['eq'],
    filterRequired: true,
  })
  storeId: number;
}
```
2. look at the GraphQL Endpoint:
![image](https://user-images.githubusercontent.com/3898144/108624948-67a0f080-7448-11eb-8b00-e6ba415d20cf.png)

**Expected behavior**
Only the permitted comparisons should really be offered.

**Desktop (please complete the following information):**
 - Node Verson 15.2.0
 - Nestjs-query Version 7.5.2

Thanks for your work!

**Comment by doug-martin** [comment]
@renehauck this makes sense in the case of a single comparison, I think that would be an easy change, but I'm not sure it is the correct one.

I'm curious in the case that you limit it to for example 3 comparisons (`eq`, `neq`, and `between`) would you expect `and`/`or` to be exposed then? Or, would a new option to disable comparison expressions be preferable in that case?

I have thought about `eq`, `neq` etc at `BooleanExpression`s while groupings (`and`/`or`) to be `ComparisonExpression`s to be separate. I can see in either case one would want to specify allowing `and`/`or` to be turned off. For example allowing the following  cases

```ts
{
  or: [{id: { between: { lower: 1, upper: 5} }, { id: { eq: 10 }}]
}
```

```ts
{
  or: [{id: { eq: 1 }}, { id: { eq: 10 }}]
}
```

I could see both being valid, which leads me to think that exposing an option to disable `ComparisonExpression`s would be more useful. Thoughts?

Thank you for the feature request and feedback!

**Comment by renehauck** [comment]
Yes, I like that idea. :smiley:  I am not a database expert, but is it possible for a hacker to overload the database by generating massive and/or lists and querying them?
Maybe it would also make sense to put a length limit on the and/or groupings or?

**Comment by doug-martin** [comment]
@renehauck I was just starting to look at this and it dawned on me that I think this option probably belongs at the resolver level. The reason being, you could have one field that only allows `eq` while others that allow all filter comparisons. In that case I would think it should be an option you can specify. I'll need to investigate further to see which way provides the cleanest implementation. 

I think your right on the massive and/or lists especially on a large table or if there are missing indexes that cause the query to take a long time, it may not overload it but could certainly cause performance impacts. Given that I think exposing the ability to limit what you can do like disabling and/or expressions, or the max number of filter conditions may make sense. Maybe we can use the complexity plugin to handle that? 

Either way I have some research to do!

**Comment by renehauck** [comment]
Many thanks for the work! Perhaps a small tip from me. As a workaround, I have now written an interceptor that simply checks the and/or array, which is enough for me personally. Maybe you could solve it with class-validator, as you already wrote on resolver level.  I can also experiment a bit in the days to come if you like.

**Comment by doug-martin** [comment]
This is now available in `v0.24.0`. Thanks again for the feature request!
